src/freenas/usr/local/bin/truenas-initrd.py

```diff
@@ -243,24 +297,28 @@ def update_zfs_module_config(root, database):
         if root != "/":
             sys.path.insert(0, os.path.join(root, "usr/lib/python3/dist-packages"))
 
+        initialize_readonly_state(root)
+
         from middlewared.service_exception import CallError
         from middlewared.utils.db import FREENAS_DATABASE, query_config_table, query_table
         from middlewared.utils.gpu import get_gpus
 
         database = args.database or os.path.join(root, FREENAS_DATABASE[1:])
 
+        set_readonly(root, readonly_desired_value=False)
```

Comment:
Why change readonly state if update is not required?

Reply:
I can specify this in different functions so this gets executed when we actually want to just turn it off, the function should handle this nicely now